src/sqlfluff/core/rules/base.py

```diff
@@ -484,149 +586,107 @@ def _eval(self, context: RuleContext) -> EvalResultType:
 
     def crawl(
         self,
-        segment,
-        ignore_mask,
-        dialect,
-        parent_stack=None,
```

Comment:
Several of these parameters were already stored on `RuleContext`. Eliminated these as individual parameters; now they are stored and passed using `RuleContext`. This was basically old tech debt from before `RuleContext` was added.